lib/matplotlib/markers.py

```diff
@@ -363,11 +391,101 @@ def get_alt_transform(self):
+        else:
+            return self._alt_transform.frozen()
 
     def get_snap_threshold(self):
         return self._snap_threshold
 
+    def get_user_transform(self):
+        """Return user supplied part of marker transform."""
+        if self._user_transform is not None:
+            return self._user_transform.frozen()
+
+    def transformed(self, transform: Affine2D):
+        """
+        Return new marker with combined transformation.
+
+        Parameters
+        ----------
+        transform : Affine2D, default: None
+            Transform will be combined with current user supplied transform.
+        """
+        new_marker = MarkerStyle(self)
+        if new_marker._user_transform is not None:
+            new_marker._user_transform += transform
+        else:
+            new_marker._user_transform = transform
+        return new_marker
+
+    def rotated(self, deg=None, rad=None):
+        """
+        Return new marker rotated by specified angle.
+
+        Parameters
+        ----------
+        deg : float, default: None
+            Use this parameter to specify rotation angle in degrees.
+
+        rad : float, default: None
+            Use this parameter to specify rotation angle in radians.
+
+        .. note:: You must specify exactly one of deg or rad.
+        """
+        if not ((deg is None) ^ (rad is None)):
+            raise ValueError("Exactly one of deg or rad shall be used.")
+
+        new_marker = MarkerStyle(self)
+        if new_marker._user_transform is None:
+            new_marker._user_transform = Affine2D()
+
+        if deg is not None:
+            new_marker._user_transform.rotate_deg(deg)
+        if rad is not None:
+            new_marker._user_transform.rotate(rad)
+
+        return new_marker
+
+    def scaled(self, sx, sy=None):
+        """
+        Return new marker scaled by specified scale factors.
+
+        If *sy* is None, the same scale is applied in both the *x*- and
+        *y*-directions.
+
+        Parameters
+        ----------
+        sx : float
+            *X*-direction scaling factor.
+        sy : float, default: None
+            *Y*-direction scaling factor.
+        """
+        if sy is None:
+            sy = sx
+
+        new_marker = MarkerStyle(self)
+        _transform = new_marker._user_transform or Affine2D()
+        new_marker._user_transform = _transform.scale(sx, sy)
+        return new_marker
+
+    def translated(self, tx, ty):
+        """
+        Return new marker translated by tx and ty.
+
+        Parameters
+        ----------
+        tx : float
+            Coordinate for translation in *x*-direction.
+        ty : float
+            Coordinate for translation in *y*-direction.
+        """
+        new_marker = MarkerStyle(self)
+        _transform = new_marker._user_transform or Affine2D()
+        new_marker._user_transform = _transform.translate(tx, ty)
+        return new_marker
```

Comment:
Looks like this code needs a test too.

Reply:
Sorry for the missing tests, `translated` is now removed.